lib/colocated-babel-plugin.js

```diff
@@ -15,6 +19,53 @@ module.exports = function (babel) {
     name: 'ember-cli-htmlbars-colocation-template',
 
     visitor: {
+      Program(path, state) {
+        let allAddedImports = {};
+
+        state._colocationEnsureImport = (exportName, moduleName) => {
+          let addedImports = (allAddedImports[moduleName] = allAddedImports[moduleName] || {});
+
+          if (addedImports[exportName]) return addedImports[exportName];
+
+          let importDeclarations = path.get('body').filter((n) => n.type === 'ImportDeclaration');
+
+          let preexistingImportDeclaration = importDeclarations.find(
+            (n) => n.get('source').get('value').node === moduleName
+          );
+
+          if (preexistingImportDeclaration) {
+            let importSpecifier = preexistingImportDeclaration
+              .get('specifiers')
+              .find(({ node }) => {
+                return exportName === 'default'
+                  ? t.isImportDefaultSpecifier(node)
+                  : node.imported && node.imported.name === exportName;
+              });
+
+            if (importSpecifier) {
+              addedImports[exportName] = importSpecifier.node.local;
+            }
+          }
+
+          if (!addedImports[exportName]) {
+            let uid = path.scope.generateUidIdentifier(
+              exportName === 'default' ? moduleName : exportName
+            );
+            addedImports[exportName] = uid;
+
+            let newImportSpecifier =
+              exportName === 'default'
+                ? t.importDefaultSpecifier(uid)
+                : t.importSpecifier(uid, t.identifier(exportName));
+
+            let newImport = t.importDeclaration([newImportSpecifier], t.stringLiteral(moduleName));
+            path.unshiftContainer('body', newImport);
+          }
+
+          return addedImports[exportName];
```

Comment:
Doesn't this suffer from the same problem that https://github.com/ember-cli/babel-plugin-htmlbars-inline-precompile/pull/357 fixed?